src/qibo/quantum_info/hamming_weight.py

```diff
@@ -0,0 +1,260 @@
+        measurements (list, optional): list of measurements gates :class:`qibo.gates.M`.
+            Defaults to ``None``.
+        nshots (int, optional): number of shots used for sampling the measurements.
+            Defaults to :math:`1000`.
+        engine (str, optional): engine to use in the execution of the
+            :class:`qibo.backends.HammingWeightBackend`. It accepts ``"numpy"``, ``"numba"``,
+            ``"cupy"``, and ``"cuquantum"``. If ``None``, defaults to the corresponding engine
+            from the current backend. Defaults to ``None``.
+    """
+
+    def __init__(
+        self,
+        state,
+        weight: int,
+        nqubits: int,
+        measurements=None,
+        nshots: int = 1000,
+        engine=None,
+    ):  # pylint: disable=too-many-arguments
+
+        backend = HammingWeightBackend(engine)
+        QuantumState.__init__(self, state, backend)
+        MeasurementOutcomes.__init__(self, measurements, backend)
+
+        self.measurements = measurements
+        self.nqubits = nqubits
+        self.nshots = nshots
+        self.weight = weight
+        self.n_choose_k = int(binom(self.nqubits, self.weight))
+
+        self._state = state
+        self._samples = None
+        self._probs = None
+        self._frequencies = None
+        self._measurement_gate = None
+        self._repeated_execution_frequencies = None
+        self._measurement_gate = self.measurement_gate
+
+    def symbolic(self, decimals: int = 5, cutoff: float = 1e-10, max_terms: int = 20):
+        """Dirac notation representation of the state in the computational basis.
+
+        Args:
+            decimals (int, optional): Number of decimals for the amplitudes.
+                Defaults to :math:`5`.
+            cutoff (float, optional): Amplitudes with absolute value smaller than the
+                cutoff are ignored from the representation. Defaults to  :math:`1e-10`.
+            max_terms (int, optional): Maximum number of terms to print. If the state
+                contains more terms they will be ignored. Defaults to :math:`20`.
+
+        Returns:
+            str: String representing the state in the computational basis.
+        """
+        terms = self.backend.calculate_symbolic(
+            self._state, self.nqubits, self.weight, decimals, cutoff, max_terms
+        )
+        return " + ".join(terms)
+
+    def state(self, numpy: bool = False):
+        """Tensor representation of ``state`` as a backend tensor in the subspace
+        of fixed Hamming weight.
+
+        Args:
+            numpy (bool, optional): If ``True`` the returned tensor will be a ``numpy`` array,
+                otherwise it will follow the backend tensor type.
+                Defaults to ``False``.
+
+        Returns:
+            ndarray: State in the computational basis of the subspace
+            with fixed Hamming weight ordered in lexicographical order.
+        """
+        return super().state(numpy=numpy)
```

Comment:
I believe these can just be inherited from `QuantumState`.

Reply:
@AlejandroSopena

Reply:
The `symbolic` method is a custom implementation to work specifically with this backend. It requires passing the `weight` to `backend.calculate_symbolic`. As for `state`, it differs only in the docstring. I have removed it from this class and added a note in the `QuantumState` method to clarify this.